test/integration/authz_v2_test.go

```diff
@@ -0,0 +1,399 @@
+		athenzDomainList, err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().List(metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if len(athenzDomainList.Items) != 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	assert.Nil(t, err, "Athenz domain should be removed without errors")
+
+	// Validate Authorization Policy has not been removed
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, noop, noop)
+
+	// Cleanup
+	deleteServiceObjects(t, e.Services)
+}
+
+// Initial: Existing service with annotation, AP
+// Input Actions: delete service with annotation
+// Output: AP deleted
+func TestDeleteServiceShouldDeleteAuthorizationPolicy(t *testing.T) {
+	// Initial set up
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Delete services
+	applyServices(t, e.Services, delete)
+
+	// Validate Athenz Domain has been removed
+	namespace := e.Services[0].Namespace
+	err := wait.PollImmediate(time.Second, time.Second*5, func() (bool, error) {
+		servicesList, err := framework.Global.K8sClientset.CoreV1().Services(namespace).List(metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		if len(servicesList.Items) != 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+
+	// Validate Authorization Policy has been removed
+	verifyAuthorizationPolicyIsRemoved(t, e.AuthorizationPolicies[0])
+
+	assert.Nil(t, err, "Authorization policy list not empty")
+
+	// Cleanup
+	applyAthenzDomain(t, e.AD, delete)
+}
+
+// Initial: Existing service with annotation, AP
+// Input Actions: delete service annotation
+// Output: AP deleted
+func TestRemoveAnnotationFromServiceShouldDeleteAuthorizationPolicy(t *testing.T) {
+	// Initial set up
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Remove the annotation from the service
+	updatedValues := fixtures.GetBasicRbacV2Case(&fixtures.RbacV2Modifications{
+		ModifyServices: [][]func(service *v1.Service){
+			{
+				func(service *v1.Service) {
+					service.Annotations = make(map[string]string)
+				},
+			},
+		},
+	})
+	// Apply the updated service
+	applyServices(t, updatedValues.Services, update)
+
+	// Verify that the Authorization policy is removed
+	verifyAuthorizationPolicyIsRemoved(t, e.AuthorizationPolicies[0])
+
+	// Cleanup resources
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Initial: Existing service with annotation, AP
+// Input Actions: add an athenz role member with expiry
+// Output: after member expired, AP update with this member removal
+func TestUpdateAuthorizationPolicyRemovingExpiredMembers(t *testing.T) {
+	// Initial set up
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Update Athenz Domain with an expiring member
+	// Get updated fixtures
+	newUserName := "user.bar"
+	value := time.Now().Add(time.Second * time.Duration(30))
```

Comment:
feel like there is no need to add 30s for current time, we can just set the expiration time to current time, and reduce the time to wait for expiration to a shorter period

Reply:
We poll for the authorization policies every second for 5 times (seconds). If the expiry time is set to time now, we cannot be sure if the update for the Authorization Policy did not go through or if the member role expired before the athenz domain is processed when this test case fails. I tested out various delays and found that 30seconds and 1 minute reliably works.

Reply:
ok got it.